File: VB/CalcWeightedAverage/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraReports.UI
' ...

Namespace CalcWeightedAverage
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub lblSummaryPrice_SummaryGetResult(ByVal sender As Object, ByVal e As SummaryGetResultEventArgs) Handles lblSummaryPrice.SummaryGetResult
			Dim sum1 As Double = 0
			Dim sum2 As Double = 0
			For Each s As String In e.CalculatedValues
				Dim vv() As String = s.Split("|"c)
				sum1 += Double.Parse(vv(0))
				sum2 += Double.Parse(vv(1))
			Next s
			e.Result = sum1 / sum2
			e.Handled = True
		End Sub

	End Class
End Namespace

File: CS/CalcWeightedAverage/XtraReport1.cs
using DevExpress.XtraReports.UI;
// ...

namespace CalcWeightedAverage {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void lblSummaryPrice_SummaryGetResult(object sender, SummaryGetResultEventArgs e) {
            double sum1 = 0;
            double sum2 = 0;
            foreach (string s in e.CalculatedValues) {
                string[] vv = s.Split('|');
                sum1 += double.Parse(vv[0]);
                sum2 += double.Parse(vv[1]);
            }
            e.Result = sum1 / sum2;
            e.Handled = true;
        }

    }
}
